src/release/main.d

```diff
@@ -354,6 +354,64 @@ ActionList clearReleaseNotes ( )
 
 /*******************************************************************************
 
+    Prepares the actions for a major release
+
+    Params:
+        con = connection object
+        repo = repo object
+        tags = array of tags for this repo
+        major_version = version to release
+
+    Returns:
+        prepared ActionList object with all refs/actions required for this
+        release
+
+*******************************************************************************/
+
+ActionList prepareMajorRelease ( ref HTTPConnection con, ref Repository repo,
+                                 Version[] tags, Version major_version )
+{
+    import release.gitHelper;
+
+    import std.algorithm : find;
+    import std.range : empty;
+    import std.format : format;
+
+    auto current_branch = SemVerBranch(getCurrentBranch());
+
+    auto list = makeRelease(major_version, current_branch.toString);
+
+    auto next_major = current_branch;
+    next_major.major++;
+
+    auto branches = getBranches(con, repo, major_version, false);
+
+    auto next_major_rslt = branches.find(next_major);
+
+    if (next_major_rslt.empty)
+    {
+        list.local_actions ~= LocalAction(format("git branch %s", next_major),
+                                          format("Create next major branch %s",
+                                                 next_major));
+    }
+
+    list.local_actions ~= LocalAction(format("git checkout %s", next_major),
+                                      format("Checkout next major branch %s",
+                                             next_major));
+
+    list ~= clearReleaseNotes();
+
+    list.affected_refs ~= next_major.toString;
+
+    list.local_actions ~= LocalAction(format("git checkout %s", current_branch),
+                                      format("Checkout original branch %s",
+                                             current_branch));
+    return list;
+}
```

Comment:
This whole function doesn't make a lot of sense tbh. I suggested to not support major release by `neptune-release` exactly because it is mostly manual effort with very few automation. For example, this function does not rebase major branch before releasing.

Maybe it should just blindly tag / aggregate release notes on current branch trusting developer to do all necessary preparations?